src/main/java/com/github/dockerjava/core/command/ListImagesCmdImpl.java

```diff
@@ -48,4 +49,22 @@ public ListImagesCmd withFilters(String filter) {
         return this;
     }
 
+    @Override
+    public String toString() {
+        return new StringBuilder("images ").append(showAll ? "--all=true" : "")
+                .append(filters != null ? "--filters " + filters : "")
+                .append(imageNameFilter != null ? "--filter " + imageNameFilter : "").toString();
+    }
+
+    @Override
+    public ListImagesCmd withImageNameFilter(String imageNameFilter) {
+        checkNotNull(imageNameFilter, "image name filter not specified");
```

Comment:
No, annotation is source level, real check is runtime check. 
If you want disable filter, then you can't nullify it as i understand. So probably checkNotNull and nonnull annotation could be just removed.